coderd/coderd.go

```diff
@@ -478,12 +480,14 @@ type API struct {
 
 	derpServer *derp.Server
 
-	Handler             chi.Router
+	Auditor             atomic.Pointer[audit.Auditor]
```

Comment:
Couldn't the same methodology be applied here, but with `Load()`? It's easier to abuse, sure, but a feature implementation couldn't change after you obtain the pointer. We could add a context argument wrapping a pointer load to prevent this style of abuse. Thoughts?

Reply:
That requires the API struct to get into the business of knowing which features need to be atomically updated together, so they can be grouped into a single struct for `Load`ing.  With `Get()` the code that needs the features can define what it needs.

Exposing the pointer directly allows any random bit of code to call `Store()` on it, which is certainly undesirable.

The other nice feature of `Get()` and the feature Server is that it works for code that is not an API handler.  With this design, any code that needs to interact with features will have to have a reference to the whole API, instead of just the small feature Server interface.

Reply:
That's under the assumption that Enterprise features are dynamically working together and making assumptions of that in AGPL code, right? While not impossible, I'm tempted to wait for that to emerge naturally rather than prepare for it now. We could mitigate that problem in a simple, low-cost way by adding a mutex to load the values.

> Exposing the pointer directly allows any random bit of code to call Store() on it, which is certainly undesirable.

I don't understand this point because there are many areas of the code that we don't abstract to fight bad actors. We'll discuss it live!

> With this design, any code that needs to interact with features will have to reference the whole API instead of just the small feature Server interface.

Couldn't any feature just reference the atomic pointer instead? That way it wouldn't need the entire API.

Reply:
> I don't understand this point because there are many areas of the code that we don't abstract to fight bad actors. We'll discuss it live!

I think the spirit of the atomic pointer is correct we should just wrap it and remove the harmful parts (namely `Store()` as @spikecurtis pointed out)...In my opinion it's not so much about bad actors as some unwitting developer unintentionally abusing the API in a dubious way.